Fix golangci-lint errors with go-atlassian v1.4.4

### Affected Resources

- atlassian_jira_issue_type_screen_scheme
- atlassian_jira_screen_scheme


### Terraform CLI and Provider Versions

provider: v0.1.0

### Terraform Configuration

```terraform
na
```


### Expected Behavior

No errors should be present when executing the command `golangci-lint run`.

### Actual Behavior

```console
golangci-lint run
internal/provider/data_source_jira_issue_type_screen_scheme.go:129:80: cannot use ([]int literal) (value of type []int) as *"github.com/ctreminiom/go-atlassian/pkg/infra/models".ScreenSchemeParamsScheme value in argument to d.p.jira.Issue.Type.ScreenScheme.Gets (typecheck)
	issueTypeScreenScheme, res, err := d.p.jira.Issue.Type.ScreenScheme.Gets(ctx, []int{issueTypeScreenSchemeId}, 0, 50)
	                                                                              ^
internal/provider/data_source_jira_screen_scheme.go:136:61: cannot use ([]int literal) (value of type []int) as *"github.com/ctreminiom/go-atlassian/pkg/infra/models".ScreenSchemeParamsScheme value in argument to d.p.jira.Screen.Scheme.Gets (typecheck)
	screenScheme, res, err := d.p.jira.Screen.Scheme.Gets(ctx, []int{screenSchemeId}, 0, 50)
	                                                           ^
internal/provider/resource_jira_issue_type_screen_scheme.go:205:87: cannot use ([]int literal) (value of type []int) as *"github.com/ctreminiom/go-atlassian/pkg/infra/models".ScreenSchemeParamsScheme value in argument to r.p.jira.Issue.Type.ScreenScheme.Gets (typecheck)
	issueTypeScreenSchemeDetails, res, err := r.p.jira.Issue.Type.ScreenScheme.Gets(ctx, []int{issueTypeScreenSchemeId}, 0, 50)
	                                                                                     ^
internal/provider/resource_jira_screen_scheme.go:215:64: cannot use ([]int literal) (value of type []int) as *"github.com/ctreminiom/go-atlassian/pkg/infra/models".ScreenSchemeParamsScheme value in argument to r.p.jira.Screen.Scheme.Gets (typecheck)
	resScreenScheme, res, err := r.p.jira.Screen.Scheme.Gets(ctx, []int{screenSchemeId}, 0, 50)
	                                                              ^
```

### Steps to Reproduce

Execute `golangci-lint run`

### How much impact is this issue causing?

High

### Logs

_No response_

### Additional Information

Relates #151 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/internal/provider/data_source_jira_issue_type_screen_scheme.go
+++ b/internal/provider/data_source_jira_issue_type_screen_scheme.go
@@ -6,6 +6,7 @@ import (
 	"strconv"
 
 	jira "github.com/ctreminiom/go-atlassian/jira/v3"
+	"github.com/ctreminiom/go-atlassian/pkg/infra/models"
 	"github.com/hashicorp/terraform-plugin-framework/datasource"
 	"github.com/hashicorp/terraform-plugin-framework/diag"
 	"github.com/hashicorp/terraform-plugin-framework/path"
@@ -109,7 +110,7 @@ func (d *jiraIssueTypeScreenSchemeDataSource) Configure(ctx context.Context, req
 }
 
 func (d *jiraIssueTypeScreenSchemeDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
-	tflog.Debug(ctx, "Reading issue type screen scheme")
+	tflog.Debug(ctx, "Reading issue type screen scheme data source")
 
 	var newState jiraIssueTypeScreenSchemeDataSourceModel
 	resp.Diagnostics.Append(req.Config.Get(ctx, &newState)...)
@@ -117,7 +118,7 @@ func (d *jiraIssueTypeScreenSchemeDataSource) Read(ctx context.Context, req data
 		return
 	}
 	tflog.Debug(ctx, "Loaded issue type screen scheme config", map[string]interface{}{
-		"issueTypeScreenScheme": fmt.Sprintf("%+v", newState),
+		"readConfig": fmt.Sprintf("%+v", newState),
 	})
 
 	issueTypeScreenSchemeId, err := strconv.Atoi(newState.ID.ValueString())
@@ -125,8 +126,11 @@ func (d *jiraIssueTypeScreenSchemeDataSource) Read(ctx context.Context, req data
 		resp.Diagnostics.AddAttributeError(path.Root("id"), "Unable to parse value of \"id\" attribute.", "Value of \"id\" attribute can only be a numeric string.")
 		return
 	}
+	options := &models.ScreenSchemeParamsScheme{
+		IDs: []int{issueTypeScreenSchemeId},
+	}
 
-	issueTypeScreenScheme, res, err := d.p.jira.Issue.Type.ScreenScheme.Gets(ctx, []int{issueTypeScreenSchemeId}, 0, 50)
+	issueTypeScreenScheme, res, err := d.p.jira.Issue.Type.ScreenScheme.Gets(ctx, options, 0, 1)
 	if err != nil {
 		var resBody string
 		if res != nil {
@@ -135,12 +139,6 @@ func (d *jiraIssueTypeScreenSchemeDataSource) Read(ctx context.Context, req data
 		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to get issue type screen scheme, got error: %s\n%s", err, resBody))
 		return
 	}
-	tflog.Debug(ctx, "Retrieved issue type screen scheme from API state", map[string]interface{}{
-		"issueTypeScreenScheme": fmt.Sprintf("%+v", issueTypeScreenScheme.Values[0]),
-	})
-
-	newState.Name = types.String{Value: issueTypeScreenScheme.Values[0].Name}
-	newState.Description = types.String{Value: issueTypeScreenScheme.Values[0].Description}
 
 	issueTypeMappings, res, err := d.p.jira.Issue.Type.ScreenScheme.Mapping(ctx, []int{issueTypeScreenSchemeId}, 0, 50)
 	if err != nil {
@@ -151,9 +149,12 @@ func (d *jiraIssueTypeScreenSchemeDataSource) Read(ctx context.Context, req data
 		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to get issue type screen scheme mappings, got error: %s\n%s", err, resBody))
 		return
 	}
-	tflog.Debug(ctx, "Retrieved issue type screen scheme mappings from API state", map[string]interface{}{
-		"issueTypeMappings": fmt.Sprintf("%+v", issueTypeMappings.Values[0]),
+	tflog.Debug(ctx, "Retrieved issue type screen scheme from API state", map[string]interface{}{
+		"readApiState": fmt.Sprintf("%+v, Mappings:%+v", issueTypeScreenScheme.Values[0], issueTypeMappings.Values[0]),
 	})
+
+	newState.Name = types.String{Value: issueTypeScreenScheme.Values[0].Name}
+	newState.Description = types.String{Value: issueTypeScreenScheme.Values[0].Description}
 	var mappings []jiraIssueTypeScreenSchemeMapping
 	for _, m := range issueTypeMappings.Values {
 		mappings = append(mappings, jiraIssueTypeScreenSchemeMapping{

--- a/internal/provider/data_source_jira_screen_scheme.go
+++ b/internal/provider/data_source_jira_screen_scheme.go
@@ -6,6 +6,7 @@ import (
 	"strconv"
 
 	jira "github.com/ctreminiom/go-atlassian/jira/v3"
+	"github.com/ctreminiom/go-atlassian/pkg/infra/models"
 	"github.com/hashicorp/terraform-plugin-framework/datasource"
 	"github.com/hashicorp/terraform-plugin-framework/diag"
 	"github.com/hashicorp/terraform-plugin-framework/path"
@@ -116,7 +117,7 @@ func (d *jiraScreenSchemeDataSource) Configure(ctx context.Context, req datasour
 }
 
 func (d *jiraScreenSchemeDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
-	tflog.Debug(ctx, "Reading screen scheme")
+	tflog.Debug(ctx, "Reading screen scheme data source")
 
 	var newState jiraScreenSchemeDataSourceModel
 	resp.Diagnostics.Append(req.Config.Get(ctx, &newState)...)
@@ -124,7 +125,7 @@ func (d *jiraScreenSchemeDataSource) Read(ctx context.Context, req datasource.Re
 		return
 	}
 	tflog.Debug(ctx, "Loaded screen scheme config", map[string]interface{}{
-		"screenScheme": fmt.Sprintf("%+v", newState),
+		"readConfig": fmt.Sprintf("%+v", newState),
 	})
 
 	screenSchemeId, err := strconv.Atoi(newState.ID.ValueString())
@@ -133,7 +134,10 @@ func (d *jiraScreenSchemeDataSource) Read(ctx context.Context, req datasource.Re
 		return
 	}
 
-	screenScheme, res, err := d.p.jira.Screen.Scheme.Gets(ctx, []int{screenSchemeId}, 0, 50)
+	options := &models.ScreenSchemeParamsScheme{
+		IDs: []int{screenSchemeId},
+	}
+	screenScheme, res, err := d.p.jira.Screen.Scheme.Gets(ctx, options, 0, 1)
 	if err != nil {
 		var resBody string
 		if res != nil {
@@ -142,7 +146,7 @@ func (d *jiraScreenSchemeDataSource) Read(ctx context.Context, req datasource.Re
 		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to get screen scheme, got error: %s\n%s", err, resBody))
 	}
 	tflog.Debug(ctx, "Retrieved screen scheme from API state", map[string]interface{}{
-		"screenScheme": fmt.Sprintf("%+v", screenScheme.Values[0]),
+		"readApiState": fmt.Sprintf("%+v", screenScheme.Values[0]),
 	})
 
 	newState.Name = types.String{Value: screenScheme.Values[0].Name}
@@ -154,6 +158,6 @@ func (d *jiraScreenSchemeDataSource) Read(ctx context.Context, req datasource.Re
 		Edit:    types.Int64{Value: int64(screenScheme.Values[0].Screens.Edit)},
 	}
 
-	tflog.Debug(ctx, "Storing screen scheme info into the state")
+	tflog.Debug(ctx, "Storing screen scheme into the state")
 	resp.Diagnostics.Append(resp.State.Set(ctx, &newState)...)
 }

--- a/internal/provider/resource_jira_issue_type_screen_scheme.go
+++ b/internal/provider/resource_jira_issue_type_screen_scheme.go
@@ -131,15 +131,15 @@ func (*jiraIssueTypeScreenSchemeResource) ImportState(ctx context.Context, req r
 }
 
 func (r *jiraIssueTypeScreenSchemeResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
-	tflog.Debug(ctx, "Creating issue type screen scheme")
+	tflog.Debug(ctx, "Creating issue type screen scheme resource")
 
 	var plan jiraIssueTypeScreenSchemeResourceModel
 	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
-	tflog.Debug(ctx, "Loaded issue type screen scheme configuration", map[string]interface{}{
-		"issueTypeScreenSchemeConfig": fmt.Sprintf("%+v", plan),
+	tflog.Debug(ctx, "Loaded issue type screen scheme plan", map[string]interface{}{
+		"createPlan": fmt.Sprintf("%+v", plan),
 	})
 
 	issueTypeMappings := []*models.IssueTypeScreenSchemeMappingPayloadScheme{}
@@ -152,11 +152,9 @@ func (r *jiraIssueTypeScreenSchemeResource) Create(ctx context.Context, req reso
 
 	createRequestPayload := models.IssueTypeScreenSchemePayloadScheme{
 		Name:              plan.Name.ValueString(),
+		Description:       plan.Description.ValueString(),
 		IssueTypeMappings: issueTypeMappings,
 	}
-	tflog.Debug(ctx, "Generated request payload", map[string]interface{}{
-		"issueTypeScreenScheme": fmt.Sprintf("%+v", createRequestPayload),
-	})
 
 	newIssueTypeScreenScheme, res, err := r.p.jira.Issue.Type.ScreenScheme.Create(ctx, &createRequestPayload)
 	if err != nil {
@@ -167,30 +165,18 @@ func (r *jiraIssueTypeScreenSchemeResource) Create(ctx context.Context, req reso
 		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to create issue type screen scheme, got error: %s\n%s", err, resBody))
 		return
 	}
-	tflog.Debug(ctx, "Created issue type screen scheme", map[string]interface{}{
-		"issueTypeScreenScheme": fmt.Sprintf("%+v", newIssueTypeScreenScheme),
-	})
+	tflog.Debug(ctx, "Created issue type screen scheme")
 
 	plan.ID = types.String{Value: newIssueTypeScreenScheme.ID}
 
-	// TODO: Remove this when 'description' can be addded on create call above
-	// https://github.com/ctreminiom/go-atlassian/issues/131
-	res, err = r.p.jira.Issue.Type.ScreenScheme.Update(ctx, newIssueTypeScreenScheme.ID, plan.Name.ValueString(), plan.Description.ValueString())
-	if err != nil {
-		var resBody string
-		if res != nil {
-			resBody = res.Bytes.String()
-		}
-		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to add issue type screen scheme description, got error: %s\n%s", err, resBody))
-		return
-	}
-
-	tflog.Debug(ctx, "Storing issue type screen scheme info into the state")
+	tflog.Debug(ctx, "Storing issue type screen scheme into the state", map[string]interface{}{
+		"createNewState": fmt.Sprintf("%+v", plan),
+	})
 	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
 }
 
 func (r *jiraIssueTypeScreenSchemeResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
-	tflog.Debug(ctx, "Reading issue type screen scheme")
+	tflog.Debug(ctx, "Reading issue type screen scheme resource")
 
 	var state jiraIssueTypeScreenSchemeResourceModel
 	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
@@ -198,11 +184,14 @@ func (r *jiraIssueTypeScreenSchemeResource) Read(ctx context.Context, req resour
 		return
 	}
 	tflog.Debug(ctx, "Loaded issue type screen scheme from state", map[string]interface{}{
-		"issueTypeScreenSchemeState": fmt.Sprintf("%+v", state),
+		"readState": fmt.Sprintf("%+v", state),
 	})
 
 	issueTypeScreenSchemeId, _ := strconv.Atoi(state.ID.ValueString())
-	issueTypeScreenSchemeDetails, res, err := r.p.jira.Issue.Type.ScreenScheme.Gets(ctx, []int{issueTypeScreenSchemeId}, 0, 50)
+	options := &models.ScreenSchemeParamsScheme{
+		IDs: []int{issueTypeScreenSchemeId},
+	}
+	issueTypeScreenSchemeDetails, res, err := r.p.jira.Issue.Type.ScreenScheme.Gets(ctx, options, 0, 1)
 	if err != nil {
 		var resBody string
 		if res != nil {
@@ -235,21 +224,21 @@ func (r *jiraIssueTypeScreenSchemeResource) Read(ctx context.Context, req resour
 	state.IssueTypeMappings = mappings
 
 	tflog.Debug(ctx, "Storing issue type screen scheme into the state", map[string]interface{}{
-		"newState": fmt.Sprintf("%+v", state),
+		"readNewState": fmt.Sprintf("%+v", state),
 	})
 	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
 }
 
 func (r *jiraIssueTypeScreenSchemeResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
-	tflog.Debug(ctx, "Updating issue type screen scheme")
+	tflog.Debug(ctx, "Updating issue type screen scheme resource")
 
 	var plan jiraIssueTypeScreenSchemeResourceModel
 	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
-	tflog.Debug(ctx, "Loaded issue type screen scheme configuration", map[string]interface{}{
-		"issueTypeScreenSchemeConfig": fmt.Sprintf("%+v", plan),
+	tflog.Debug(ctx, "Loaded issue type screen scheme plan", map[string]interface{}{
+		"updatePlan": fmt.Sprintf("%+v", plan),
 	})
 
 	var state jiraIssueTypeScreenSchemeResourceModel
@@ -258,7 +247,7 @@ func (r *jiraIssueTypeScreenSchemeResource) Update(ctx context.Context, req reso
 		return
 	}
 	tflog.Debug(ctx, "Loaded issue type screen scheme from state", map[string]interface{}{
-		"issueTypeScreenSchemeState": fmt.Sprintf("%+v", state),
+		"updateState": fmt.Sprintf("%+v", state),
 	})
 
 	err := r.updateNameAndDescription(ctx, &plan, &state)
@@ -289,12 +278,12 @@ func (r *jiraIssueTypeScreenSchemeResource) Update(ctx context.Context, req reso
 
 	plan.ID = types.String{Value: state.ID.ValueString()}
 
-	tflog.Debug(ctx, "Storing issue type screen scheme info into the state")
+	tflog.Debug(ctx, "Storing issue type screen scheme into the state")
 	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
 }
 
 func (r *jiraIssueTypeScreenSchemeResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
-	tflog.Debug(ctx, "Deleting issue type screen scheme")
+	tflog.Debug(ctx, "Deleting issue type screen scheme resource")
 
 	var state jiraIssueTypeScreenSchemeResourceModel
 	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
@@ -309,10 +298,10 @@ func (r *jiraIssueTypeScreenSchemeResource) Delete(ctx context.Context, req reso
 		if res != nil {
 			resBody = res.Bytes.String()
 		}
-		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to delete issue type screen scheme, got error: %s\n%s", err.Error(), resBody))
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to delete issue type screen scheme, got error: %s\n%s", err, resBody))
 		return
 	}
-	tflog.Debug(ctx, "Removed issue type screen scheme from API state")
+	tflog.Debug(ctx, "Deleted issue type screen scheme from API state")
 
 	// If a Resource type Delete method is completed without error, the framework will automatically remove the resource.
 }

--- a/internal/provider/resource_jira_issue_type_screen_scheme_test.go
+++ b/internal/provider/resource_jira_issue_type_screen_scheme_test.go
@@ -94,22 +94,22 @@ func TestAccJiraIssueTypeScreenScheme_IssueTypeMappings(t *testing.T) {
 		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccIssueTypeScreenSchemeConfig_issuetypemappings(resourceName, randomName, "10000"),
+				Config: testAccIssueTypeScreenSchemeConfig_issuetypemappings(resourceName, randomName, "foo"),
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr(resourceName, "issue_type_mappings.#", "2"),
 					resource.TestCheckResourceAttr(resourceName, "issue_type_mappings.0.issue_type_id", "default"),
 					resource.TestCheckResourceAttr(resourceName, "issue_type_mappings.0.screen_scheme_id", "1"),
-					resource.TestCheckResourceAttr(resourceName, "issue_type_mappings.1.issue_type_id", "10000"),
+					resource.TestCheckResourceAttrPair(resourceName, "issue_type_mappings.1.issue_type_id", "atlassian_jira_issue_type.foo", "id"),
 					resource.TestCheckResourceAttr(resourceName, "issue_type_mappings.1.screen_scheme_id", "1"),
 				),
 			},
 			{
-				Config: testAccIssueTypeScreenSchemeConfig_issuetypemappings(resourceName, randomName, "10001"),
+				Config: testAccIssueTypeScreenSchemeConfig_issuetypemappings(resourceName, randomName, "bar"),
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr(resourceName, "issue_type_mappings.#", "2"),
 					resource.TestCheckResourceAttr(resourceName, "issue_type_mappings.0.issue_type_id", "default"),
 					resource.TestCheckResourceAttr(resourceName, "issue_type_mappings.0.screen_scheme_id", "1"),
-					resource.TestCheckResourceAttr(resourceName, "issue_type_mappings.1.issue_type_id", "10001"),
+					resource.TestCheckResourceAttrPair(resourceName, "issue_type_mappings.1.issue_type_id", "atlassian_jira_issue_type.bar", "id"),
 					resource.TestCheckResourceAttr(resourceName, "issue_type_mappings.1.screen_scheme_id", "1"),
 				),
 			},
@@ -185,9 +185,17 @@ func testAccIssueTypeScreenSchemeConfig_defaultmapping(resourceName, name string
 	}
 }
 
-func testAccIssueTypeScreenSchemeConfig_issuetypemappings(resourceName, name, nonDefaultIssueTypeId string) string {
+func testAccIssueTypeScreenSchemeConfig_issuetypemappings(resourceName, name, nonDefaultIssueTypeSuffix string) string {
 	splits := strings.Split(resourceName, ".")
 	return fmt.Sprintf(`
+	resource "atlassian_jira_issue_type" "foo" {
+		name = "%[3]s-foo"
+	}
+
+	resource "atlassian_jira_issue_type" "bar" {
+		name = "%[3]s-bar"
+	}
+
 	resource %[1]q %[2]q {
 		name = %[3]q
 		issue_type_mappings = [
@@ -196,10 +204,10 @@ func testAccIssueTypeScreenSchemeConfig_issuetypemappings(resourceName, name, no
 				screen_scheme_id = "1"
 			},
 			{
-				issue_type_id = %[4]q 
+				issue_type_id = atlassian_jira_issue_type.%[4]s.id  
 				screen_scheme_id = "1" 
 			}
 		]
 	}
-	`, splits[0], splits[1], name, nonDefaultIssueTypeId)
+	`, splits[0], splits[1], name, nonDefaultIssueTypeSuffix)
 }

--- a/internal/provider/resource_jira_screen_scheme.go
+++ b/internal/provider/resource_jira_screen_scheme.go
@@ -23,10 +23,10 @@ type (
 	}
 
 	jiraScreenSchemeResourceModel struct {
-		ID          types.String               `tfsdk:"id"`
-		Name        types.String               `tfsdk:"name"`
-		Description types.String               `tfsdk:"description"`
-		Screens     jiraScreenSchemeTypesModel `tfsdk:"screens"`
+		ID          types.String                `tfsdk:"id"`
+		Name        types.String                `tfsdk:"name"`
+		Description types.String                `tfsdk:"description"`
+		Screens     *jiraScreenSchemeTypesModel `tfsdk:"screens"`
 	}
 
 	jiraScreenSchemeTypesModel struct {
@@ -78,6 +78,9 @@ func (*jiraScreenSchemeResource) GetSchema(_ context.Context) (tfsdk.Schema, dia
 				Type:     types.StringType,
 				Validators: []tfsdk.AttributeValidator{
 					stringvalidator.LengthAtMost(255),
+				},
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					attribute_plan_modification.DefaultValue(types.StringValue("")),
 				},
 			},
 			"screens": {
@@ -146,9 +149,6 @@ func (r *jiraScreenSchemeResource) Configure(ctx context.Context, req resource.C
 
 func (*jiraScreenSchemeResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
 	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
-	// Must initialise "singleNestedAttributes" to avoid "unhandled null values" error when calling (tfsdk.Plan).Get
-	screens := jiraScreenSchemeTypesModel{}
-	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("screens"), screens)...)
 }
 
 func (r *jiraScreenSchemeResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
@@ -159,13 +159,9 @@ func (r *jiraScreenSchemeResource) Create(ctx context.Context, req resource.Crea
 	if resp.Diagnostics.HasError() {
 		return
 	}
-	tflog.Debug(ctx, "Loaded screen scheme configuration", map[string]interface{}{
-		"screenSchemeConfig": fmt.Sprintf("%+v", plan),
+	tflog.Debug(ctx, "Loaded screen scheme plan", map[string]interface{}{
+		"createPlan": fmt.Sprintf("%+v", plan),
 	})
-
-	if plan.Description.IsUnknown() || plan.Description.IsNull() {
-		plan.Description = types.String{Value: ""}
-	}
 
 	createRequestPayload := models.ScreenSchemePayloadScheme{
 		Screens: &models.ScreenTypesScheme{
@@ -177,25 +173,23 @@ func (r *jiraScreenSchemeResource) Create(ctx context.Context, req resource.Crea
 		Name:        plan.Name.ValueString(),
 		Description: plan.Description.ValueString(),
 	}
-	tflog.Debug(ctx, "Generated request payload", map[string]interface{}{
-		"screenSchemeReq": fmt.Sprintf("%+v", createRequestPayload),
-	})
-	newScreenScheme, res, err := r.p.jira.Screen.Scheme.Create(ctx, &createRequestPayload)
+
+	screenScheme, res, err := r.p.jira.Screen.Scheme.Create(ctx, &createRequestPayload)
 	if err != nil {
 		var resBody string
 		if res != nil {
 			resBody = res.Bytes.String()
 		}
-		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to create screen scheme, got error: %s\n%s", err.Error(), resBody))
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to create screen scheme, got error: %s\n%s", err, resBody))
 		return
 	}
-	tflog.Debug(ctx, "Created screen scheme", map[string]interface{}{
-		"screenScheme": newScreenScheme.ID,
+	tflog.Debug(ctx, "Created screen scheme")
+
+	plan.ID = types.StringValue(strconv.Itoa(screenScheme.ID))
+
+	tflog.Debug(ctx, "Storing screen scheme into the state", map[string]interface{}{
+		"createNewState": fmt.Sprintf("%+v", plan),
 	})
-
-	plan.ID = types.String{Value: strconv.Itoa(newScreenScheme.ID)}
-
-	tflog.Debug(ctx, "Storing screen scheme info into the state")
 	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
 }
 
@@ -208,44 +202,48 @@ func (r *jiraScreenSchemeResource) Read(ctx context.Context, req resource.ReadRe
 		return
 	}
 	tflog.Debug(ctx, "Loaded screen scheme from state", map[string]interface{}{
-		"screenSchemeState": fmt.Sprintf("%+v", state),
+		"readState": fmt.Sprintf("%+v", state),
 	})
 
 	screenSchemeId, _ := strconv.Atoi(state.ID.ValueString())
-	resScreenScheme, res, err := r.p.jira.Screen.Scheme.Gets(ctx, []int{screenSchemeId}, 0, 50)
+	options := &models.ScreenSchemeParamsScheme{
+		IDs: []int{screenSchemeId},
+	}
+	resScreenScheme, res, err := r.p.jira.Screen.Scheme.Gets(ctx, options, 0, 1)
 	if err != nil {
 		var resBody string
 		if res != nil {
 			resBody = res.Bytes.String()
 		}
-		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to get screen scheme, got error: %s\n%s", err.Error(), resBody))
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to get screen scheme, got error: %s\n%s", err, resBody))
 		return
 	}
 	tflog.Debug(ctx, "Retrieved screen scheme from API state")
 
 	state.Name = types.String{Value: resScreenScheme.Values[0].Name}
 	state.Description = types.String{Value: resScreenScheme.Values[0].Description}
-	state.Screens = jiraScreenSchemeTypesModel{
+	state.Screens = &jiraScreenSchemeTypesModel{
 		Create:  types.Int64{Value: int64(resScreenScheme.Values[0].Screens.Create)},
 		Default: types.Int64{Value: int64(resScreenScheme.Values[0].Screens.Default)},
 		View:    types.Int64{Value: int64(resScreenScheme.Values[0].Screens.View)},
 		Edit:    types.Int64{Value: int64(resScreenScheme.Values[0].Screens.Edit)},
 	}
-	tflog.Debug(ctx, "Storing screen scheme info into the state")
+	tflog.Debug(ctx, "Storing screen scheme into the state", map[string]interface{}{
+		"readNewState": fmt.Sprintf("%+v", state),
+	})
 	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
 }
 
 func (r *jiraScreenSchemeResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
-	tflog.Debug(ctx, "Updating screen scheme")
+	tflog.Debug(ctx, "Updating screen scheme resource")
 
 	var plan jiraScreenSchemeResourceModel
 	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
-
-	tflog.Debug(ctx, "Loaded screen scheme configuration", map[string]interface{}{
-		"screenSchemeConfig": fmt.Sprintf("%+v", plan),
+	tflog.Debug(ctx, "Loaded screen scheme plan", map[string]interface{}{
+		"updatePlan": fmt.Sprintf("%+v", plan),
 	})
 
 	var state jiraScreenSchemeResourceModel
@@ -253,9 +251,8 @@ func (r *jiraScreenSchemeResource) Update(ctx context.Context, req resource.Upda
 	if resp.Diagnostics.HasError() {
 		return
 	}
-
 	tflog.Debug(ctx, "Loaded screen scheme from state", map[string]interface{}{
-		"screenSchemestate": fmt.Sprintf("%+v", state),
+		"updateState": fmt.Sprintf("%+v", state),
 	})
 
 	updateRequestPayload := models.ScreenSchemePayloadScheme{
@@ -268,22 +265,20 @@ func (r *jiraScreenSchemeResource) Update(ctx context.Context, req resource.Upda
 			Edit:    int(plan.Screens.Edit.ValueInt64()),
 		},
 	}
-	tflog.Debug(ctx, "Generated request payload", map[string]interface{}{
-		"screenSchemeReq": fmt.Sprintf("%+v", updateRequestPayload),
-	})
+
 	res, err := r.p.jira.Screen.Scheme.Update(ctx, state.ID.ValueString(), &updateRequestPayload)
 	if err != nil {
 		var resBody string
 		if res != nil {
 			resBody = res.Bytes.String()
 		}
-		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to update screen scheme, got error: %s\n%s", err.Error(), resBody))
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to update screen scheme, got error: %s\n%s", err, resBody))
 	}
 	tflog.Debug(ctx, "Updated screen scheme in API state")
 
 	plan.ID = types.String{Value: state.ID.ValueString()}
 
-	tflog.Debug(ctx, "Storing screen scheme info into the state")
+	tflog.Debug(ctx, "Storing screen scheme into the state")
 	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
 }
 
@@ -303,10 +298,10 @@ func (r *jiraScreenSchemeResource) Delete(ctx context.Context, req resource.Dele
 		if res != nil {
 			resBody = res.Bytes.String()
 		}
-		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to delete screen scheme, got error: %s\n%s", err.Error(), resBody))
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to delete screen scheme, got error: %s\n%s", err, resBody))
 		return
 	}
-	tflog.Debug(ctx, "Removed screen scheme from API state")
+	tflog.Debug(ctx, "Deleted screen scheme from API state")
 
 	// If a Resource type Delete method is completed without error, the framework will automatically remove the resource.
 }

